addons/mail/static/src/components/chat_window_manager/chat_window_manager_tests.js

```diff
@@ -1351,6 +1372,225 @@ QUnit.test('chat window: switch on TAB', async function (assert) {
     );
 });
 
+QUnit.test('chat window: scroll position conservation on folded', async function (assert) {
+    assert.expect(3);
+
+    Object.assign(this.data.initMessaging, {
+        channel_slots: {
+            channel_channel: [{
+                channel_type: 'channel',
+                id: 20,
+                is_minimized: false,
+                is_pinned: true,
+                name: "General",
+                state: 'open',
+                uuid: 'channel-20-uuid',
+            }],
+        },
+    });
+    this.data['mail.message'].records = [];
+    for(const i of Array(20).keys()) {
+        this.data['mail.message'].records.push({
+            author_id: [7, "Demo"],
+            body: "<p>body</p>",
+            channel_ids: [20],
+            date: "2019-04-20 10:00:00",
+            id: i,
+            message_type: 'comment',
+            model: 'mail.channel',
+            record_name: 'General',
+            res_id: 20,
+        });
+    }
+    await this.start();
+    await afterNextRender(() => document.querySelector(`.o_MessagingMenu_toggler`).click());
+    await afterNextRender(() =>
+        document.querySelector(`.o_NotificationList_preview`).click()
+    );
+    // Set a scroll position to chat window
+    document.querySelector(`.o_ThreadViewer_messageList`).scrollTop = 142;
+    assert.strictEqual(
+        document.querySelector(`.o_ThreadViewer_messageList`).scrollTop,
+        142,
+        "verify chat window initial scrollTop"
+    );
+
+    // fold chat window
+    await afterNextRender(() => document.querySelector('.o_ChatWindow_header').click());
+    assert.containsNone(
+        document.body,
+        ".o_ThreadViewer",
+        "chat window should be folded so no thread viewer should be present"
+    );
+
+    // unfold chat window
+    await afterNextRender(() => document.querySelector('.o_ChatWindow_header').click());
+    assert.strictEqual(
+        document.querySelector(`.o_ThreadViewer_messageList`).scrollTop,
+        142,
+        "chat window scrollTop should still be the same when chat window is unfolded"
+    );
+});
+
+QUnit.test('[technical] chat window: composer state conservation on toggle home menu when folded', async function (assert) {
```

Comment:
same remark as before: whenever a test is flagged as `[technical]`, briefly explain why in a comment in heading of test content.